### PR TITLE
feat: scg speed up

### PR DESCRIPTION
**Pull request by smousy**

### CODE REVIEW

File: client/js/Ui/windowmanager.js [hunk 1]
```diff
@@ -196,6 +197,12 @@ SCWeb.ui.WindowManager = {
 
                 self.hideActiveWindow();
                 self.windows.push(id);
+                if (self.windows.length > self.MAX_WINDOWS) {
+                    const lastWindowId = self.windows[0];
+                    self.windows.shift();
```
Comment (KovalM) [inline client/js/Ui/windowmanager.js:202]:
```suggestion
                    const lastWindowId = self.windows.shift();
```

File: client/js/Core/componentsandbox.js [hunk 2]
```diff
@@ -265,12 +265,16 @@ SCWeb.core.ComponentSandbox.prototype.updateContent = async function (contentTyp
     var self = this;
 
     if (this.is_struct && this.eventStructUpdate) {
+        const maxNumberOfTriplets = 100;
```
Comment (KovalM) [inline client/js/Core/componentsandbox.js:268]: I think we don't need this limitation because it's timely solution. We can wait for right solution

**Review by KovalM — CHANGES_REQUESTED** [review]

File: client/js/Core/componentsandbox.js (hunk 2 above)
Comment (smousy) [reply client/js/Core/componentsandbox.js:268]: let's keep this timely solution, it's really useful, maybe we can create issue?

Comment (KovalM) [reply client/js/Core/componentsandbox.js:268]: Ok, let's find optimal limitation and create issue

**Review by KovalM — CHANGES_REQUESTED** [review]

**Review by KovalM — APPROVED** [review]

File: components/scg/src/scg-struct.js [hunk 3]
```diff
@@ -1,3 +1,17 @@
+const debaunce = (func, wait) => {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-struct.js:1]:
```suggestion
const debounce = (func, wait) => {
```

File: components/scg/src/scg-struct.js [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+    let timerId;
+
+    const clear = () => {
+        clearTimeout(timerId);
+    };
+    const debaunced = (...args) => {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-struct.js:7]:
```suggestion
    const debounced = (...args) => {
```

File: components/scg/src/scg-struct.js [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+        clearTimeout(timerId);
+        timerId = setTimeout(() => func(...args), wait);
+    };
+
+    return [debaunced, clear];
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-struct.js:12]:
```suggestion
    return [debounced, clear];
```

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

**Comment by MikhailSadovsky** [comment]
Please don't forget to update changelog

**Pull request merged**